spdx/writers/xml.py

```diff
@@ -9,22 +9,185 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import xmltodict
+import xml.etree.ElementTree as ET
 
+from spdx import creationinfo, file
+from spdx.checksum import CHECKSUM_ALGORITHM_TO_XML_DICT
+from spdx.utils import NoAssert, SPDXNone
 from spdx.writers.tagvalue import InvalidDocumentError
-from spdx.writers.jsonyamlxml import Writer
 from spdx.parsers.loggers import ErrorMessages
 
 
-def write_document(document, out, validate=True):
+def add_not_empty(parent, element_name, element_value):
+    if element_value is not None:
+        ET.SubElement(parent, element_name).text = str(element_value)
+
+
+def add_checksum(parent, tag, new_checksum):
+    checksum_element = ET.SubElement(parent, tag)
+    ET.SubElement(checksum_element, 'checksumValue').text = new_checksum.value
+    algo = CHECKSUM_ALGORITHM_TO_XML_DICT.get(new_checksum.identifier) or 'checksumAlgorithm_sha1'
+    ET.SubElement(checksum_element, 'algorithm').text = algo
+
+
+def add_license(parent, tag, new_license):
+    license_text = None
+    if isinstance(new_license, NoAssert):
+        license_text = new_license.to_value()
+    elif isinstance(license_text, SPDXNone):
+        license_text = 'NONE'
+    else:
+        license_text = new_license.identifier
+    ET.SubElement(parent, tag).text = license_text
+
 
+def add_version(parent, tag, version):
+    ET.SubElement(parent, tag).text = '{}.{}'.format(version.major, version.minor)
+
+
+def write_document(document, out, validate=True):
     if validate:
         messages = ErrorMessages()
         messages = document.validate(messages)
         if messages:
             raise InvalidDocumentError(messages)
 
-    writer = Writer(document)
-    document_object = {"SpdxDocument": writer.create_document()}
+    spdx_doc_element = ET.Element('SpdxDocument')
+    doc_element = ET.SubElement(spdx_doc_element, 'Document')
+    add_not_empty(doc_element, 'comment', document.comment)
+    add_not_empty(doc_element, 'name', document.name)
+    document_describes_element = ET.SubElement(doc_element, 'documentDescribes')
+    for package in document.packages:
+        packages_element = ET.SubElement(document_describes_element, 'Package')
+        add_not_empty(packages_element, 'SPDXID', package.spdx_id)
+        add_not_empty(packages_element, 'originator', package.originator)
+        add_not_empty(packages_element, 'attributionTexts', package.attribution_text)
+        for phile in package.files:
+            file_element = ET.SubElement(packages_element, 'File')
+            add_not_empty(file_element, 'comment', phile.comment)
+            for license_in_file in phile.licenses_in_file:
+                add_license(file_element, 'licenseInfoFromFiles', license_in_file)
+            sha1 = None
+            for checksum in phile.checksums:
+                if checksum.identifier == 'SHA1':
+                    sha1 = checksum.value
+                    break
```

Comment:
There is another loop over `checksums` below that should be merged with this one

Reply:
that loop went away with the 'sha1' member.